### PR TITLE
Added canceled subscriber to handle incoming buffers after getting cancel signal

### PR DESCRIPTION
**Pull request by segabriel**
Closes https://github.com/scalecube/reactor-aeron/projects/1#card-16665969

### CODE REVIEW

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java [hunk 1]
```diff
@@ -24,6 +28,9 @@
           AtomicReferenceFieldUpdater.newUpdater(
               DefaultAeronInbound.class, CoreSubscriber.class, "destinationSubscriber");
 
+  private static final CoreSubscriber<? super DirectBuffer> CANCELED_SUBSCRIBER =
```
Comment (artem-v) [inline reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: CANCE`LL`ED_SUBSCRIBER

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java [hunk 2]
```diff
@@ -24,6 +28,9 @@
...
+      new CanceledSubscriber();
```
Comment (artem-v) [inline reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: Cance`ll`edSubscriber

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java [hunk 3]
```diff
@@ -124,10 +134,8 @@ public void request(long n) {
 
     @Override
     public void cancel() {
-      REQUESTED.set(DefaultAeronInbound.this, 0);
```
Comment (artem-v) [inline reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: Why removed line 127?

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java [hunk 4]
```diff
@@ -124,10 +134,8 @@ public void request(long n) {
...
-      // TODO think again whether re-subscribtion is allowed; or we have to should emit cancel
-      // signal upper to some conneciton shutdown hook
-      CoreSubscriber destination = DESTINATION_SUBSCRIBER.getAndSet(DefaultAeronInbound.this, null);
+      CoreSubscriber destination =
+          DESTINATION_SUBSCRIBER.getAndSet(DefaultAeronInbound.this, CANCELED_SUBSCRIBER);
```
Comment (artem-v) [inline reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: Plz add debug log: "Destination subscriber on aeron inbound has been cancelled"

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java [hunk 5]
```diff
@@ -146,4 +154,27 @@ public void subscribe(CoreSubscriber<? super DirectBuffer> destinationSubscriber
       }
     }
   }
+
+  private static class CanceledSubscriber implements CoreSubscriber<DirectBuffer> {
```
Comment (artem-v) [inline reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: Cance`ll`edSubscriber

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java [hunk 6]
```diff
@@ -146,4 +154,27 @@ public void subscribe(CoreSubscriber<? super DirectBuffer> destinationSubscriber
...
+
+    @Override
+    public void onSubscribe(Subscription s) {
+      // no-op
+    }
+
+    @Override
+    public void onNext(DirectBuffer directBuffer) {
+      LOGGER.debug("was received buffer in the already canceled subscriber, {}", directBuffer);
```
Comment (artem-v) [inline reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: I think this must be .warn()  "Received message, which will be dropped immediately due cancelled aeron inbound"

File: reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java (hunk 3 above)
Comment (segabriel) [reply reactor-aeron-core/src/main/java/reactor/aeron/DefaultAeronInbound.java]: it's redundant now - we always check the subscriber first

**Pull request merged**